src/sdk/namespace_s3.js

```diff
@@ -523,13 +538,23 @@ class NamespaceS3 {
             xattr,
             tag_count: res.TagCount,
             first_range_data: res.Body,
+            part_count: res.PartsCount,
         };
     }
 
-    _translate_error_code(err) {
+    _translate_error_code(params, err) {
         if (err.code === 'NoSuchKey') err.rpc_code = 'NO_SUCH_OBJECT';
-        if (err.code === 'NotFound') err.rpc_code = 'NO_SUCH_OBJECT';
-        if (err.code === 'PreconditionFailed') err.rpc_code = 'IF_MATCH_ETAG';
+        else if (err.code === 'NotFound') err.rpc_code = 'NO_SUCH_OBJECT';
+        else {
+            const md_conditions = params.md_conditions;
```

Comment:
Don't you need check that md_conditions is defined?
```suggestion
        else if (params.md_conditions) {
            const md_conditions = params.md_conditions;
```